zebra/rtadv.c

```diff
@@ -535,9 +536,28 @@ static int rtadv_timer(struct thread *thread)
 static void rtadv_process_solicit(struct interface *ifp)
 {
 	struct zebra_vrf *zvrf = vrf_info_lookup(ifp->vrf_id);
+	struct zebra_if *zif;
 
 	assert(zvrf);
-	rtadv_send_packet(rtadv_get_socket(zvrf), ifp);
+	zif = ifp->info;
+
+	/*
+	 * If FastRetransmit is enabled, send the RA immediately.
+	 * If not enabled but it has been more than MIN_DELAY_BETWEEN_RAS
+	 * (3 seconds) since the last RA was sent, send it now and reset
+	 * the timer to start at the max (configured) again.
+	 * If not enabled and it is less than 3 seconds since the last
+	 * RA packet was sent, set the timer for 3 seconds so the next
+	 * one will be sent with a minimum of 3 seconds between RAs.
+	 * RFC4861 sec 6.2.6
+	 */
+	if ((zif->rtadv.UseFastRexmit)
```

Comment:
Excellent, detailed comment - very helpful... but I'm not following how the math here aligns with the comment?

a) is `MaxRtrAdvInterval` guaranteed to be > `MIN_DELAY` ?
b) the comment talks about a timer, and about the elapsed time since the last RA was sent - but I don't see anyone taking or using a timestamp ?

Reply:
Based on the RFC stating minimum between RAs is 3 seconds, my assumption is that it would always be true.  Now that you point it out, this is an assumption I'll need to flesh out and do a bit more checking/testing.

Reply:
As for the question about the timer, there is a timer event that fires and the value is decremented and tested to see if  it's time to send an RA.  This function sets that remaining time value to influence when the RA is sent (or delayed.)

Reply:
In testing, if I set the ra-interval shorter than MIN_DELAY_BETWEEN_RAS and have "no ipv6 nd ra-fast-retrans" enabled, if a solicit is received, it delays for three seconds as required by the RFC before sending out the RA in response.  Note that I doubt any of our customers anywhere will enable this knob, but the conformance testers required the ability rate limit the RAs per the RFC.